Auth0/OAuth2Grant.swift

```diff
@@ -43,15 +54,18 @@ struct ImplicitGrant: OAuth2Grant {
     }
 
     func credentials(from values: [String: String], callback: @escaping (Result<Credentials>) -> Void) {
-        guard validate(responseType: self.responseType, token: values["id_token"], nonce: self.defaults["nonce"]) else {
-            return callback(.failure(error: WebAuthError.invalidIdTokenNonce))
+        let responseType = self.responseType
+        let validatorContext = IDTokenValidatorContext(authentication: authentication,
+                                                       leeway: leeway,
+                                                       nonce: self.defaults["nonce"],
+                                                       maxAge: maxAge)
+        validate(idToken: values["id_token"], for: responseType, with: validatorContext) { error in
```

Comment:
I'm having a tough time understanding what this block is doing. Can you explain? Is it failing if there is no access token in the response? I see that was the behavior before as well but seems funny to fail without one. Definitely might be something I'm misunderstanding.

Reply:
Yes, this is currently failing for `response_type = id_token` for example. Shouldn't fail, it should accept all implicit flow scenarios, which are 3 in total: the combination of `id_token` and `token`.

I suggest moving to an `&&` instead and negate the condition, something like this:
```js
if response_type.contains("token") && !values.access_token -> fail
```
Anyway, something to fix on a separate PR as this was existing behavior.

with @Widcket now. cc @cocojoe

Reply:
Will be fixed on a separate PR. Thanks for raising this.